plugins/nordic-hid/fu-nordic-hid-cfg-channel.c

```diff
@@ -419,78 +423,317 @@ fu_nordic_hid_cfg_channel_cmd_receive(FuNordicHidCfgChannel *self,
 }
 
 static gboolean
-fu_nordic_hid_cfg_channel_add_peers(FuNordicHidCfgChannel *self, GError **error)
+fu_nordic_hid_cfg_channel_is_cached_peer_connected(guint8 peer_cache_val)
+{
+	return (peer_cache_val % 2) != 0;
+}
+
+static void
+fu_nordic_hid_cfg_channel_add_peer(FuNordicHidCfgChannel *self, guint8 peer_id)
+{
+	g_autoptr(GError) error_local = NULL;
+	g_autoptr(FuNordicHidCfgChannel) peer = NULL;
+
+	peer = fu_nordic_hid_cfg_channel_new(peer_id, self);
+
+	/* ensure that the general quirk content for Nordic HID devices is applied */
+	fu_device_add_instance_id(FU_DEVICE(peer), "HIDRAW\\VEN_1915");
+
+	if (!fu_device_setup(FU_DEVICE(peer), &error_local)) {
+		g_debug("failed to discover peer 0x%02x: %s", peer_id, error_local->message);
+		return;
+	}
+
+	g_debug("peer 0x%02x discovered", peer_id);
+	fu_device_add_child(FU_DEVICE(self), FU_DEVICE(peer));
+	/* prohibit to close parent's communication descriptor */
+	fu_device_add_internal_flag(FU_DEVICE(peer), FU_DEVICE_INTERNAL_FLAG_USE_PARENT_FOR_OPEN);
+}
+
+static void
+fu_nordic_hid_cfg_channel_remove_peer(FuNordicHidCfgChannel *self, guint8 peer_id)
+{
+	GPtrArray *children = fu_device_get_children(FU_DEVICE(self));
+
+	/* Remove child device if already discovered. */
+	for (guint i = 0; i < children->len; i++) {
+		FuDevice *child_dev = g_ptr_array_index(children, i);
+		FuNordicHidCfgChannel *child = FU_NORDIC_HID_CFG_CHANNEL(child_dev);
+
+		if (child->peer_id == peer_id) {
+			fu_device_remove_child(FU_DEVICE(self), child_dev);
+			break;
+		}
+	}
+}
+
+static void
+fu_nordic_hid_cfg_channel_remove_disconnected_peers(FuNordicHidCfgChannel *self,
+						    guint8 peers_cache[PEERS_CACHE_LEN])
+{
+	for (guint i = 0; i < PEERS_CACHE_LEN; i++) {
+		guint8 peer_id = i + 1;
+
+		if (peers_cache == NULL ||
+		    !fu_nordic_hid_cfg_channel_is_cached_peer_connected(peers_cache[i])) {
+			fu_nordic_hid_cfg_channel_remove_peer(self, peer_id);
+			if (peers_cache != NULL)
+				self->peers_cache[i] = peers_cache[i];
+		}
+	}
+}
+
+static gboolean
+fu_nordic_hid_cfg_channel_index_peers_cmd(FuNordicHidCfgChannel *self,
+					  GError **error,
+					  gboolean *cmd_supported)
```

Comment:
I think you aligned in every other function that `error` is last, consider aligning it here also